### DISCUSSION TITLE
using NuGet that depends on Uno.Toolkit.WinUI gives build error on Windows target - tries to copy assets?

### DISCUSSION THREAD

**Discussion by VincentH-Net** [unoplatform/uno.extensions#1921] [Q&A] [answered]
(all on latest stable - Uno 4.10.13, Uno.Toolkit.WinUI 4.2.22)

When I create a NuGet `csharpmarkup.winui.uno.toolkit` that depends on NuGet `Uno.Toolkit.WinUI`:
`<PackageReference Include="Uno.Toolkit.WinUI" Version="4.2.22" />`
, and I include my NuGet in a Uno application, the build of the Windows target fails with messages like:

> Error MSB3030 Could not copy the file "C:\Users\dev\.nuget\packages\csharpmarkup.winui.uno.toolkit\2.3.30\lib\net7.0-windows10.0.18362\Uno.Toolkit.WinUI\Assets\BackIcon.scale-400.png" because it was not found.

It appears that the build tries to copy assets from the `Uno.Toolkit.WinUI` that is nested within `csharpmarkup.winui.uno.toolkit` - however since I also include the Uno.Toolkit.WinUI NuGet itself in the app, the assets could come from there (I assume).

How can I get the Windows target building?

(I'm unclear on whether my NuGet should "pass on" the assets of the Uno Toolkit NuGet that it depends on, or alternatively should somehow prevent that the build tries to copy these assets)

Clarification & help appreciated!

**Comment by jeromelaban** [comment]
Thanks for the report. This is tracked in https://github.com/unoplatform/uno.toolkit.ui/issues/854.

**Reply by VincentH-Net** [reply]
Thanks for the pointer.

When I add `ExcludeAssets="contentFiles"` to the Uno package reference in my own NuGet, the windows target build errors in the app reduce from:
![image](https://github.com/unoplatform/uno.extensions/assets/1872271/8dfe81fe-39ee-4c39-892b-1dbea0d2ca6e)
to:
![image](https://github.com/unoplatform/uno.extensions/assets/1872271/6ee4af5e-e697-48c5-8dc7-593e99dd8efe)

Adding `ExcludeAssets="contentFiles"` to my own package reference and / or the Uno package references in the app does not remove these remaining errors - any idea how to do that?

PS I am also using this workaround for the Windows target in both the app project and my own NuGet project:
`<WindowsAppSDKWinUI>true</WindowsAppSDKWinUI> <!-- Workaround for https://github.com/unoplatform/uno/discussions/13569 -->`

**Comment by VincentH-Net** [comment]
I lost patience and implemented a hack workaround:

For the Windows target I just copied and referenced the DLLs out of the offending Uno NuGets, instead of using the NuGets themselves. I had to leave out the `.PRI` files for the DLLs - they also reference the assets.

Note that `Uno.Extensions.Hosting.WinUI` has the same problem as `Uno.Toolkit.WinUI` (see error messages in above screenshots)

If anyone wants to use this workaround: [here is the diff](https://github.com/VincentH-Net/CSharpForMarkup/commit/af94fc59cb7412b2a9df8b7ee5b45013b7c3c0fc)

**Comment by VincentH-Net** [comment, marked as answer]
I resolved this - see [final workaround](https://github.com/unoplatform/uno.toolkit.ui/issues/854#issuecomment-1744442216)
